src/expansionmap.cpp

```diff
@@ -246,4 +247,16 @@ void ExpansionMap::GetUuidList(Object *object, std::vector<std::string> &idList)
     }
 }
 
+void ExpansionMap::GeneratePredictableIds(Object *source, Object *target)
+{
+    unsigned i = 0;
+    ArrayOfObjects targetObjects = *target->GetChildren();
```

Comment:
It is not safe not to check that the number of children is the same.

Reply:
I was assuming that a clone has the same size as its source, but now I am checking.